core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -121,180 +125,143 @@ public TableOperations underlyingOps() {
     return ops;
   }
 
-  private void checkLastOperationCommitted(String operation) {
+  protected PendingUpdate appendUpdates(PendingUpdate update) {
+    if (!(update instanceof ManageSnapshots)) {
+      checkLastOperationCommitted(update.getClass().getSimpleName());
```

Comment:
I found this might change the exception message for both type `UpdatePartitionSpec` and `StreamingDelete`, if this is a concern I can update to keep the original message.

Reply:
Is this conditional on the instanceof check just to handle the `manageSnapshots` case? If so, we could just validate that the last operation committed twice and avoid this.